### PR TITLE
Add orbit camera controls

### PR DESCRIPTION
**Pull request by Williangalvani**
OrbitControl example from Three.js ported to work with Qt and QML.

Signed-off-by: Willian Galvani <williangalvani@gmail.com>

### CODE REVIEW

File: glcode.js [hunk 1]
```diff
@@ -1,18 +1,33 @@
 Qt.include("three.js")
+Qt.include("controls.js")
+Qt.include("math.js")
```
Comment (patrickelectric) [inline glcode.js]: This include isn't necessary.

File: glcode.js [hunk 2]
```diff
@@ -1,18 +1,33 @@
...
+
 var camera, scene, renderer;
 var cube;
 var mesh;
+var controls;
 
 function initializeGL(canvas) {
-    camera = new THREE.PerspectiveCamera(27, canvas.innerWidth/canvas.innerHeight, 1, 4000);
-    camera.position.z = 2800;
+    camera = new THREE.PerspectiveCamera(27, 640/480, 1, 11000);
```
Comment (patrickelectric) [inline glcode.js]: what is the problem to use canvas.inner ?

File: glcode.js [hunk 3]
```diff
@@ -1,18 +1,33 @@
...
     scene = new THREE.Scene();
     renderer = new THREE.Canvas3DRenderer({canvas: canvas, antialias: true, devicePixelRatio: canvas.devicePixelRatio});
     renderer.setClearColor(0x707070);
     renderer.setSize(canvas.width, canvas.height);
     renderer.gammaInput = true;
     renderer.gammaOutput = true;
+    scene.rotation.x = -Math.PI/2;
+
+    controls = new THREE.OrbitControls( camera, canvas);
+    console.log(controls);
```
Comment (patrickelectric) [inline glcode.js]: remove log

File: glcode.js [hunk 4]
```diff
@@ -78,18 +93,28 @@ function drawLine(pos) {
 var lastDate = 0;
 function paintGL(canvas) {
     var time = Date.now()*0.001;
-    scene.rotation.x = -Math.PI/4;
-    scene.rotation.z = time*0.5
+    controls.update(); // required if controls.enableDamping = true, or if controls.autoRotate = true
```
Comment (patrickelectric) [inline glcode.js]: Add comment before line, some people dont have HD displays.

File: glcode.js [hunk 5]
```diff
@@ -78,18 +93,28 @@ function drawLine(pos) {
...
+    controls.onMouseDown(mouse);
+}
+
+function onReleased(mouse){
+    controls.onMouseUp(mouse);
+}
+
+function onMouseWheel(mouse){
+    controls.onMouseWheel(mouse);
+}
```
Comment (patrickelectric) [inline glcode.js]: add empty last line.

File: main.qrc [hunk 6]
```diff
@@ -3,5 +3,7 @@
         <file>main.qml</file>
         <file>glcode.js</file>
         <file>three.js</file>
+        <file>math.js</file>
```
Comment (patrickelectric) [inline main.qrc]: Isn't necessary.

File: three.js [hunk 7]
```diff
@@ -2417,12 +2430,13 @@ THREE.Vector3.prototype = {
 		return this;
 	},
 
-	setFromMatrixColumn: function ( index, matrix ) {
-		
-		var offset = index * 4;
+	//willian:
```
Comment (patrickelectric) [inline three.js]: Can remove this line, git blame.

**Review by patrickelectric — CHANGES_REQUESTED** [review]

File: glcode.js (hunk 1 above)
Comment (Williangalvani) [reply glcode.js]: done

File: main.qrc (hunk 6 above)
Comment (Williangalvani) [reply main.qrc]: done

File: glcode.js (hunk 5 above)
Comment (Williangalvani) [reply glcode.js]: done

File: three.js (hunk 7 above)
Comment (Williangalvani) [reply three.js]: done

File: glcode.js (hunk 4 above)
Comment (Williangalvani) [reply glcode.js]: done

File: glcode.js (hunk 3 above)
Comment (Williangalvani) [reply glcode.js]: done

File: glcode.js (hunk 2 above)
Comment (Williangalvani) [reply glcode.js]: my bad, I was debugging because the aspectRatio was not being set properly.
Fixed.

**Pull request merged**